### PR TITLE
Fix: throw error message instead of the error object when submitting an donation made with Stripe Element gateway

### PR DESCRIPTION
**Pull request by glaubersilva**
Resolves [GIVE-345]

## Description

This PR fixes the error message that was being displayed in certain cases as `[object Object]` when a donation made with Stripe Element gateway was submitted. Now we are displaying a proper string to indicate the error instead of an object.

## Affects

The Stripe Element gateway

## Visuals

image

## Testing Instructions

**Credit Card:**

1. Create a donation and miss any of CC #, expiry, or CVV and press Donate;
2. The new friendly error message will show above the Donate button.

**Google Pay:**

1. Enable Google Pay in your Stripe Dashboard;
2. Create a donation and press Donate, a modal should be opened;
3. Close the Google modal;
4. The new friendly error message will show above the Donate button.

To set up Google Pay in your Chrome for Desktop, check out this link: https://givewp.com/documentation/add-ons/stripe-gateway/stripe-apple-pay-google-pay/

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-345]: https://stellarwp.atlassian.net/browse/GIVE-345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/PaymentGateways/Gateways/Stripe/StripePaymentElementGateway/stripePaymentElementGateway.tsx [hunk 1]
```diff
@@ -8,6 +8,8 @@ import {
 import {Elements, PaymentElement, useElements, useStripe} from '@stripe/react-stripe-js';
 import {applyFilters} from '@wordpress/hooks';
 import type {Gateway, GatewaySettings} from '@givewp/forms/types';
+import {Simulate} from 'react-dom/test-utils';
```
Comment (jonwaldstein) [inline src/PaymentGateways/Gateways/Stripe/StripePaymentElementGateway/stripePaymentElementGateway.tsx:11]: What's this import for?

File: src/PaymentGateways/Gateways/Stripe/StripePaymentElementGateway/stripePaymentElementGateway.tsx [hunk 2]
```diff
@@ -119,7 +121,13 @@ const stripePaymentElementGateway: StripeGateway = {
         const {error: submitError} = await this.elements.submit();
 
         if (submitError) {
-            throw new Error(submitError);
+            throw new Error(
+                sprintf(
+                    __('Invalid Payment Data. Error Details: %s (code: %s)', 'give'),
+                    submitError.message,
+                    submitError.code
```
Comment (jonwaldstein) [inline src/PaymentGateways/Gateways/Stripe/StripePaymentElementGateway/stripePaymentElementGateway.tsx:128]: Are we sure the submitError object always has these properties?

Comment (glaubersilva) [reply src/PaymentGateways/Gateways/Stripe/StripePaymentElementGateway/stripePaymentElementGateway.tsx:133]: Resolved in 2a194133519344890d04d84bfd9f4d4a791dde7e

**Comment by glaubersilva** [comment]
@jonwaldstein This is ready for re-review.

**Review by jonwaldstein — APPROVED** [review]
Thanks for the updates, looks good 👍

**Pull request merged**